### DISCUSSION TITLE
Strange phenomenon when hotkey meet LineEdit

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#3961] [Q&A] [answered]
(Rust, Slint1.3.0, win11)

The logic of my software is that:
* Presse the shift+f shortcut
* The search window is displayed
* Enter what you want to search for in the LineEdit.

But when I press shift+f, an 'f' is automatically typed in the LineEdit, like:

https://github.com/slint-ui/slint/assets/46142827/a816b87a-ea1a-434f-b8a5-987e24f6400c

This is strange and doesn't appear in previous versions  (<1.3.0).

I won't provide the code for now because it's complicated, but I can take the time to provide a simplified code if needed.

**Comment by ogoffart** [comment]
I don't exactly understand what the problem is. Is it that it types 'f' instead of 'F' ?
I can't think of any recent change that could have broken that.

**Reply by Horbin-Magician** [reply]
I used global_hotkey to register shift+f as a shortcut for displaying the window, and theoretically no matter how many times I press shift+f, there shouldn't be anything typed in the LineEdit.

However, when shift+f is pressed to make the window appear, LineEdit is typed with an 'f' (which is abnormal). After that, no matter how much times I press shift+f, nothing new appears in LineEdit (which is normal)

I think it has something to do with the handling of keyboard events, was there any exception that caused the keyboard event to be captured incorrectly during the first appearance of the window?

**Reply by ogoffart** [reply, marked as answer]
This may be related to changed in winit as we upgraded to winit 0.29

**Reply by Horbin-Magician** [reply]
Thanks a lot for your reply.
